### PR TITLE
Introduce general /modules REST route

### PR DESCRIPTION
**Pull request by felixarntz**
## Summary

Addresses issue #409 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: includes/Core/REST_API/REST_Routes.php [hunk 1]
```diff
@@ -343,6 +343,22 @@ private function get_routes() {
 					),
 				)
 			),
+			new REST_Route(
+				'modules',
+				array(
+					array(
+						'methods'             => WP_REST_Server::READABLE,
+						'callback'            => function( WP_REST_Request $request ) {
+							$modules = $this->modules->get_available_modules();
```
Comment (aaemnnosttv) [inline includes/Core/REST_API/REST_Routes.php]:
```suggestion
                            $modules = array_map(
                                array( $this, 'prepare_module_data_for_response' ),
                                $this->modules->get_available_modules()
                            );
```

File: includes/Core/REST_API/REST_Routes.php [hunk 2]
```diff
@@ -343,6 +343,22 @@ private function get_routes() {
...
+							return new WP_REST_Response( array_values( array_map( array( $this, 'prepare_module_data_for_response' ), $modules ) ) );
```
Comment (aaemnnosttv) [inline includes/Core/REST_API/REST_Routes.php]:
```suggestion
                            return new WP_REST_Response( array_values( $modules ) );
```

File: includes/Core/REST_API/REST_Routes.php [hunk 3]
```diff
@@ -343,6 +343,22 @@ private function get_routes() {
...
+						},
+						'permission_callback' => $can_authenticate,
+					),
+				),
+				array(
+					'schema' => $this->get_module_schema(),
```
Comment (aaemnnosttv) [inline includes/Core/REST_API/REST_Routes.php]: Should this be the same for returning a collection of modules as the schema used for returning a single module?

File: includes/Core/REST_API/REST_Routes.php (hunk 2 above)
Comment (felixarntz) [reply includes/Core/REST_API/REST_Routes.php]: @aaemnnosttv Why this change?

File: includes/Core/REST_API/REST_Routes.php (hunk 3 above)
Comment (felixarntz) [reply includes/Core/REST_API/REST_Routes.php]: Per the way WP core does it, they provide the same schema for both collections and singles.

**Comment by felixarntz** [comment]
@aaemnnosttv PR is updated.

Aside: One of the e2e tests temporarily failed on this (rerunning had it pass). Might be worth looking into:

Screenshot 2019-08-26 at 12 45 34

**Comment by aaemnnosttv** [comment]
> One of the e2e tests temporarily failed on this (rerunning had it pass). Might be worth looking into.

Yes that failed for me earlier as well. I added a fix for that in #440

**Pull request merged**